src/names.c

```diff
@@ -1145,6 +1154,80 @@ void insert_spec(nametab_t *tab, tree_t spec, spec_kind_t kind,
    *p = s;
 }
 
+void set_impl_label_proc_cnt(tree_t t, nametab_t *tab)
+{
+   if (t == NULL) {
+      tab->top_scope->lbl_cnts.proc = 0;
+      return;
+   }
+
+   assert (tree_kind(t) == T_ENTITY);
+
+   const int nstmts = tree_stmts(t);
+   int cnt = 0;
+   ident_t prefix = ident_new("_P");
+   for (int i = 0; i < nstmts; i++) {
+      tree_t stmt = tree_stmt(t, i);
+
+      // Entity statements for sure have labels since they have been implicitly
+      // labelled
+      if (ident_starts_with(tree_ident(stmt), prefix))
+         cnt++;
+   }
+   tab->top_scope->lbl_cnts.proc = cnt;
+}
+
+int* find_impl_label_loop_cnt(nametab_t *tab)
```

Comment:
It's local to this file so it should be static, but it's only used in one place so why not inline it directly in the function below?

Reply:
I added `static inline`, I was trying to keep the function below simple.

Reply:
I think it's easier to read as one function because you can read the whole thing linearly from top to bottom without jumping around. It's not a particularly long function and the logic is duplicated anywhere else.

Reply:
Done, moved only to the function.